lib/internal/modules/esm/module_job.js

```diff
@@ -92,6 +93,21 @@ class ModuleJob {
         this.module.instantiate();
       }
     } catch (e) {
+      if (e.message.includes('does not provide an export named')) {
+        const rootURL = this.module.url;
+        const childSpecifier = e.message.match(/'(.*?)'/)[1];
+        const childURL = new URL(childSpecifier, rootURL);
+        const format = await this.loader.getFormat(childURL.href);
+        if (format === 'commonjs') {
+          e.message = `The requested module '${childSpecifier}' is expected ` +
+            'to be of type CommonJS which does not support named exports. ' +
+            'CommonJS modules can be imported by importing the default.';
```

Comment:
I feel like it’s still worth mentioning which name was attempted to be loaded, and I’m afraid that “can be imported by importing the default” is a bit technical here (i.e. it would be nice to have a more concrete solution in the error message, even if it’s just an example like ``CommonJS modules can be imported through `import mod from 'module-name'` ``)